backend/src/packages/auth/auth.service.ts

```diff
@@ -18,6 +23,35 @@ class AuthService {
     return this.userService.create(userRequestDto);
   }
 
+  public async verifyLoginCredentials({
+    email,
+    password,
+  }: UserSignInRequestDto): Promise<UserSignInResponseDto | null> {
+    const user = await this.userService.findByEmail(email);
+
+    if (!user) {
+      throw new UsersError({
+        status: 404,
+        message: ExceptionMessage.USER_NOT_FOUND,
+      });
+    }
+
+    const isSamePassword = await encrypt.compare(password, user.passwordHash);
+
+    if (!isSamePassword) {
+      throw new UsersError({
+        status: 401,
+        message: ExceptionMessage.INCORRECT_PASSWORD,
+      });
```

Comment:
for errors related to auth we throw AuthError, also it's better to return incorrect credentials error message instead of pointing out what's wrong, this is for security purpose
```suggestion
      throw new AuthError({
        message: ExceptionMessage.INCORRECT_CREDENTIALS,
      });
```

Reply:
Fixed this!